Sources/BitcoinKit/Wallet/BitcoinCom/BitcoinComService+WalletTransactionProvider.swift

```diff
@@ -0,0 +1,153 @@
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+extension BitcoinComService: TransactionProvider {
+    // GET API: reload utxos
+    public func reload(addresses: [Address], completion: (([Transaction]) -> Void)?) {
+        let parameter: String = "[" + addresses.map { "\"\($0.cashaddr)\"" }.joined(separator: ",") + "]"
+        let urlString = baseUrl + "address/transactions/\(parameter)"
+        let url = URL(string: urlString.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)!)!
+
+        let task = URLSession.shared.dataTask(with: url) { [weak self] data, _, _ in
+            guard let data = data else {
+                completion?([])
+                return
+            }
+            guard let response = try? JSONDecoder().decode([[BitcoinComTransaction]].self, from: data) else {
+                completion?([])
+                return
+            }
+            self?.userDefaults.set(data, forKey: UserDefaultsKey.transactions.rawValue)
+            self?.userDefaults.synchronize()
+            completion?(response.joined().asTransactions())
+        }
+        task.resume()
+    }
+
+    // List utxos
+    public func list() -> [Transaction] {
+        guard let data = userDefaults.data(forKey: UserDefaultsKey.transactions.rawValue) else {
+            print("data is  nil")
+            return []
+        }
+
+        guard let response = try? JSONDecoder().decode([[BitcoinComTransaction]].self, from: data) else {
+            print("data cannot be decoded to response")
+            return []
+        }
+        return response.joined().asTransactions()
+    }
+
+}
+
+private extension Sequence where Element == BitcoinComTransaction {
+    func asTransactions() -> [Transaction] {
+        return compactMap { $0.asTransaction() }
+    }
+}
+
+// MARK: - GET Transactions
+private struct BitcoinComTransaction: Codable {
+    let txid: String
+    let version: UInt32
+    let locktime: UInt32
+    let vin: [TxIn]
+    let vout: [TxOut]
+    let blockhash: String
+    let blockheight: Int
+    let valueOut: Decimal
+    let size: Int
+    let valueIn: Decimal
+    let fees: Decimal
+
+    func asTransaction() -> Transaction? {
+        var inputs: [TransactionInput] = []
+        var outputs: [TransactionOutput] = []
+        for txin in vin {
+            guard let input = txin.asTransactionInput() else { return nil }
+            inputs.append(input)
+        }
+        for txout in vout {
+            guard let output = txout.asTransactionOutput() else { return nil }
+            outputs.append(output)
+        }
+        return Transaction(version: version, inputs: inputs, outputs: outputs, lockTime: locktime)
+    }
+}
+
+private struct TxIn: Codable {
+    let txid: String
+    let vout: UInt32
+    let sequence: UInt32
+    let scriptSig: ScriptSig
+    // let addr: String
```

Comment:
Why are they comment out? Please comment the reason somewhere.

Reply:
These are part of API response, but not used.

So I commented them out.